frontend/app/services/SubscriptionService.scala

```diff
@@ -123,4 +125,14 @@ class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String], val
   def createSubscription(memberId: MemberId, joinData: JoinForm, customerOpt: Option[Stripe.Customer]): Future[SubscribeResult] = {
     zuora.request(Subscribe(memberId, customerOpt, tierPlanRateIds(joinData.plan), joinData.name, joinData.deliveryAddress))
   }
+
+  def getPaymentSummary(memberId: MemberId): Future[PaymentSummary] = {
+    for {
+      subscription <- getSubscriptionStatus(memberId)
+      invoiceItems <- zuora.query[InvoiceItem](s"SubscriptionId='${subscription.current}'")
+    } yield {
+      val sortedInvoiceItems = sortInvoiceItems(invoiceItems)
```

Comment:
The `sortInvoiceItems()` method has been extracted here for testing purposes, but obviously is a bit small. I'd be tempted to combine `PaymentSummaryTest` & the "sort invoice items by charge number ascending" test into a single test that does slightly more work, and change the method to be:

```
def paymentSummaryFor(items: Seq[InvoiceItem]): PaymentSummary = {
  val sortedInvoiceItems=items.sortBy(_.chargeNumber)
  PaymentSummary(sortedInvoiceItems.last, sortedInvoiceItems.dropRight(1))
}
```